### ISSUE TITLE
Compile-time errors with posix-win32/{calc, qhsmtst}

### ISSUE THREAD

**Issue by FedericoSerafini** [QuantumLeaps/qpc-examples#7]
Hi,

I am facing issues in compiling the calc and qhsmtst posix-win32 example projects.

# posix-win32/calc

Trying to compile with "make" I encountered the following error:

```
 -I../../../ports/posix -I../../../include -I../../../ports/posix   bsp.c -o build/bsp.o
In file included from ../../../include/qpc.h:38,
                 from bsp.c:34:
../../../ports/posix/qp_port.h:48:30: error: unknown type name ‘pthread_cond_t’
   48 | #define QACTIVE_OS_OBJ_TYPE  pthread_cond_t
      |                              ^~~~~~~~~~~~~~
../../../include/qp.h:792:5: note: in expansion of macro ‘QACTIVE_OS_OBJ_TYPE’
  792 |     QACTIVE_OS_OBJ_TYPE osObject;
      |     ^~~~~~~~~~~~~~~~~~~
make: *** [Makefile:278: build/bsp.o] Error 1
```

I was able to address this defining the QV environment variable
but I am not sure if it is the right thing to do.

---

# posix-win32/qhsmtst

Trying to compile with "make" I encountered the following error:

```
In file included from ../../../src/qf/qep_hsm.c:37:
../../../include/qp_pkg.h:47:5: error: unknown type name ‘QF_EPOOL_TYPE_’
   47 |     QF_EPOOL_TYPE_ ePool_[QF_MAX_EPOOL];
      |     ^~~~~~~~~~~~~~
make: *** [Makefile:247: build/qep_hsm.o] Error 1
```

I see that the definition of the type depends on QP_IMPL but I don't know
how to proceed.

Am I missing something?

Thanks for your attention,
Regards,

  Federico

**Comment by quantum-leaps** [comment]
Hi Federico,
This is a bug in `qpc/ports/posix/qp_port.h`, which is missing `#include <pthreadd.h>`. Thank you for reporting.

A quick workaround is to use the single-thread port, which was the default in the previous versions. To do shit, please call make as follows:

`make QV=1`

The bug fix has been pushed to the qpc repo:

https://github.com/QuantumLeaps/qpc/blob/master/ports/posix/qp_port.h

So, if you like to use the multi-threaded port, please update your `qpc/ports/posix/qp_port.h` file. The file jus adds the `#include` and is fully compatible with QP/C 8.0.0.

BTW, the multi-threaded POSIX port in QP/C++ 8.0.0 is correct.

--MMS

**Comment by FedericoSerafini** [comment]
Thanks,

this solves the issues with `calc` but the compliation error for `qhsmtst` is still there.

**Comment by FedericoSerafini** [comment]
Thank you!

**Comment by quantum-leaps** [comment]
The issues have been corrected and pushed to the qpc/qpcpp repos. The changes will be officially released in QP/C/C++ 8.0.1.
--MMS
